src/stk/_internal/construction_state/molecule_state/placements_summary/placements_summary.py

```diff
@@ -205,9 +207,24 @@ def _with_functional_group_edges(
         )
         edge_ids = functional_group_edges.values()
         functional_group_edges_ = zip(functional_groups, edge_ids)
-        for functional_group, edge_id in functional_group_edges_:
-            self._edge_functional_groups[edge_id].append(
-                functional_group.with_ids(id_map)
+        try:
+            for functional_group, edge_id in functional_group_edges_:
+                self._edge_functional_groups[edge_id].append(
+                    functional_group.with_ids(id_map)
+                )
+        except IndexError:
```

Comment:
Can you try:

```python
except IndexError as error:
    error.add_note("the conent of your string here")
    raise
```

-- lemme know if the output looks better

Reply:
It does look better!